File: server/src/main/java/com/dut/healthme/service/CustomerService.java
package com.dut.healthme.service;

import com.dut.healthme.dto.request.UpdateCustomerInfoRequest;
import com.dut.healthme.dto.response.AccountInfo;
import com.dut.healthme.dto.response.CustomerInfoResponse;
import com.dut.healthme.entity.Account;
import com.dut.healthme.entity.enums.HealthGoal;
import org.springframework.web.multipart.MultipartFile;

public interface CustomerService {
    CustomerInfoResponse getCustomerInfo(Long accountId);

    CustomerInfoResponse updateHealthGoal(Long accountId, HealthGoal healthGoalRequest);

    CustomerInfoResponse updateActivityIndex(Long accountId, short activityIndex);

    CustomerInfoResponse updateInfo(Account account, UpdateCustomerInfoRequest request);

    AccountInfo updateAvatar(Account account, MultipartFile avatar);

    String getCustomerAddress(Account account);

    String updateAddress(Account account, String address);
}

File: web/src/services/customerService.js
import apiClient from './apiClient';

const customerService = {
    getCustomerInfo: async (accountId) => {
        try {
            const response = await apiClient.get(`/customer/${accountId}`);
            return response.data;
        } catch (error) {
            console.error('Error fetching customer info:', error);
        }
    },
    updateHealthGoal: async (accountId, healthGoal) => {
        try {
            const response = await apiClient.put(`/customer/${accountId}/health-goal`, { healthGoal })
            return response.data
        } catch (error) {
            console.error('Error updating health goal:', error)
            throw error;
        }
    },
    updateActivityIndex: async (accountId, activityIndex) => {
        try {
            const response = await apiClient.put(`/customer/${accountId}/activity-index/${activityIndex}`)
            return response.data
        } catch (error) {
            console.error('Error updating activity index:', error)
            throw error
        }
    },
    getCustomerAddress: async () => {
        try {
            const response = await apiClient.get(`/customer/address`)
            return response.data
        } catch (error) {
            console.error('Error updating activity index:', error)
            throw error
        }
    },
    updateCustomerAddress: async (customerAddress) => {
        try {
            const response = await apiClient.put('/customer/address', customerAddress)
            return response.data
        } catch (error) {
            console.error('Error updating customer address:', error)
        }
    }
};

export default customerService;

